compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.h

```diff
@@ -23,12 +25,25 @@ struct AMDAIEOptions {
   // Path to Vitis installation directory.
   std::string vitisInstallDir;
 
-  // Dump to stdout system commands used during compilation
-  bool showInvokedCommands;
+  // Dump system commands used during compilation
+  bool showInvokedCommands{false};
 
   // Use the legacy chess compiler.
-  bool useChess;
+  bool useChess{false};
+
+  // Print IR after all MLIR passes run in aie2xclbin (to stderr).
+  bool aie2xclbinPrintIrAfterAll{false};
+
+  // Print IR before all MLIR passes run in aie2xclbin (to stderr).
+  bool aie2xclbinPrintIrBeforeAll{false};
+
+  // Disable theading in MLIR passes in aie2xclbin.
+  bool aie2xclbinDisableTheading{false};
```

Comment:
This should probably be on by default if you want to print this.

Reply:
I know it's necessary if you're printing at the module scope


```
iree-opt --mlir-print-ir-after-all --mlir-print-ir-module-scope  test.mlir
error: IR print for module scope can't be setup on a pass-manager without disabling multi-threading first.
```

Do you thin there is something particular to this case which means we should disable threading even in the non module scope case? I wanted to make it as close the the upstream behaviour as possible.

Reply:
In general its better to avoid threading. Makes the output more determininstic and easy to diff and such. If we are printing logs, there is no real need to be efficient. We are trying to debug anyway

Reply:
If I change this to 

```
bool aie2xclbinDisableTheading{false};
```

then mlir-aie compilation will be single threaded by default, irrespective of whether we are printing anything or not. 
For this reason I think it is better to leave it as 

```
bool aie2xclbinDisableTheading{false};
```

and let the user toggle it, exactly the same way as the user would toggle the MLIR --disable-threading, --print-ir-before-all, etc flags (in any MLIR project). 
 
I could set aie2xclbinDisableTheading to true only if 

 1) user has not specified its value
 2) user has enabled printing
 
 ? I guess that's possible, but starts to get complicated...